integration/integration_test.go

```diff
@@ -2420,6 +2418,10 @@ func testDiscoveryRecovers(t *testing.T, suite *integrationTestSuite) {
 
 		// add proxy as a backend to the load balancer
 		lb.AddBackend(*utils.MustParseAddr(net.JoinHostPort(Loopback, strconv.Itoa(proxyReverseTunnelPort))))
+		// add proxy as a backend second time to prevent situations where there are only 2 backend and internal
+		// reverse tunnel dialer makes 2 calls to connect to the tunnel (ProxyPing, factual Dial call). In that
+		// situations the second lb backend reverse tunnel address is actually never dialed.
+		lb.AddBackend(*utils.MustParseAddr(net.JoinHostPort(Loopback, strconv.Itoa(proxyReverseTunnelPort))))
```

Comment:
The previous load balance setup was based two endpoints [A, B] where connection to B proxy was expected. 
After introducing reverse tunnel dialling based on proxy.Ping response the test case flow always to tries to enstablish connection to  B endpoint with Ping data obtained from proxy A. 
This change added additional B' endpoint thus the `lb` endpoints setup look like [A, B,B'] 

> Is the feature going to work when you can't count on that?

This is temporary implementation allowing to support old and new proxy setup. Also current tsh command uses ping call to obtain information about the proxy before establishing connection to the proxy. So for example current `tsh db connect` `tsh ssh`  command  implementation won't work in this setup where A,B proxy endpoints uses different addressed for DB,SSH port because `tsh db connect and `tsh ssh` commands will try to establish db connection to proxy B based on data obtained from proxy A ping response.

Reply:
As a mentioned in previous comment this is a side effect introduced by the "Ping" call during reverse tunnel dialling https://github.com/gravitational/teleport/pull/7524/files#diff-4de41f38187182c881df29c3abd9c19d2c41e213d9f7d3816f3b395ab75392ddR260

The `testDiscovery` test configures loadbalacne with two teleport proxies with different setup. Loadbalance does the sequential endpoint picking thus ping response data from proxy A are used to dial proxy B.

Reply:
Could this be a problem for customers doing an upgrade?

Reply:
I have prepared a separate PR with fix for that https://github.com/gravitational/teleport/pull/8106